### PR TITLE
Cache ImageMagick by version number and update to 6.9.5-5

### PR DESCRIPTION
**Pull request by clayton-shopify**
For review @nickhoffman 

I thought it would be easier to update ImageMagick if we included the version number in the name of the cache file. Also I bumped to the latest ImageMagick 6 release.

### CODE REVIEW

**Comment by nickhoffman** [comment]
Including the version in the cache file name is great, but it means that we'll need a way of deleting old versions periodically, otherwise they'll pile up.

File: bin/compile [hunk 1]
```diff
@@ -10,12 +10,12 @@ BUILD_DIR=$1
 CACHE_DIR=$2
 VENDOR_DIR="$BUILD_DIR/vendor"
 INSTALL_DIR="$VENDOR_DIR/imagemagick"
-CACHE_FILE="$CACHE_DIR/imagemagick.tar.gz"
-USE_CACHE_FILE='yes'
+IMAGE_MAGICK_VERSION="${IMAGE_MAGICK_VERSION:-6.9.5-5}"
```
Comment (nickhoffman) [inline bin/compile:13]: :+1:

Out of curiousity, do you know what this type of string expansion is called?

**Comment by clayton-shopify** [comment]
> we'll need a way of deleting old versions periodically, otherwise they'll pile up

That did occur to me. I guess we could just delete `imagemagick*.tar.gz` in the case where the right cache file wasn't found.

**Comment by nickhoffman** [comment]
That sounds good to me!

**Comment by nickhoffman** [comment]
:ship:

File: bin/compile (hunk 1 above)
Comment (clayton-shopify) [inline bin/compile:13]: Parameter substitution with a default value I guess? http://tldp.org/LDP/abs/html/parameter-substitution.html

**Pull request merged**